### ISSUE TITLE
Headers in docs should be clickable

### ISSUE THREAD

**Issue by witek1902** [micronaut-projects/micronaut-core#723]
Add link to all headers. 
```
<a href="#..."></a>
````
Providing links to specific parts of the documentation will be much easier.

**Comment by gijsleussink** [comment]
@witek1902, agreed. How can we make this happen? Is the documentation at https://docs.micronaut.io/latest/guide/index.html rendered by AsciiDoctor? Then `sectlinks` is an option, as mentioned in https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#section-titles

**Comment by witek1902** [comment]
@gijsleussink -> https://asciidoctor.org/docs/user-manual/#anchors
